todo-list/src/hooks/todos/useTodo.ts

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQuery } from "react-query";
+import { addTodos, fetchTodosByUserId } from "../../api/todo";
+import { Todo } from "../../types/todos/todos.type";
+
+export function useTodos() {
+  const userId = localStorage.getItem("userId");
+  const queryResult = useQuery<Todo[], Error>(
+    ["todos", userId],
+    () => fetchTodosByUserId(Number(userId)),
+    {
+      enabled: !!userId,
+      retry: 3,
+      onError: (error) => {
+        console.error("Error fetching todos:", error);
+      },
+    },
+  );
+
+  const { data: todos, error, isLoading, isError, isSuccess } = queryResult;
+  return {
+    todos,
+    error,
+    isLoading,
+    isError,
+    isSuccess,
+    refetch: queryResult.refetch,
+  };
+}
+
+export function useAddTodo() {
+  const mutation = useMutation<Todo, Error, { title: string; completed: boolean }>({
+    mutationFn: async ({ title, completed }) => {
+      const userId = localStorage.getItem("userId");
+      if (!userId) {
+        throw new Error("No userId found in localStorage");
+      }

```

Comment:
Consider a more explicit checking based on the return type of `getItem` function.

`!userId` can return false if `getItem` return an empty string